api/db-cdr/generate-cdr/build-search-all-events.sh

```diff
@@ -457,4 +459,199 @@ FROM \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v
+         AND questionnaire_response_id IS NOT NULL
+     )
+     AND observation_source_concept_id IN (
+       SELECT DISTINCT concept_id
+       FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\` c
+       JOIN (
+         SELECT CAST(id AS STRING) AS id
+         FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\`
+         WHERE concept_id IN (1585710)
+           AND domain_id = 'SURVEY' ) a ON (c.path LIKE CONCAT('%', a.id, '.%'))
+       WHERE domain_id = 'SURVEY'
+         AND type = 'PPI'
+         AND subtype = 'QUESTION'
+     )
+   )
+   SELECT *
+   FROM lifestyle_survey_remove_participants lsrp
+   WHERE se.person_id = lsrp.person_id AND se.entry_date = lsrp.observation_date AND se.concept_id = lsrp.observation_source_concept_id
+ )"
+
+##########################################################
+# Delete Basics entries where all questions have null questionnaire_response_id
+##########################################################
+echo "delete Basics entries where all questions have null questionnaire_response_id"
+bq --quiet --project_id="$BQ_PROJECT" query --nouse_legacy_sql \
+"DELETE
+ FROM \`$BQ_PROJECT.$BQ_DATASET.cb_search_all_events\` se
+ WHERE EXISTS (
+   WITH lifestyle_survey_remove_participants AS (
+     SELECT person_id, observation_date, observation_source_concept_id, questionnaire_response_id
+     FROM \`$BQ_PROJECT.$BQ_DATASET.R2022Q4R3.observation\`
+     WHERE person_id IN (
+         SELECT DISTINCT person_id
+         FROM \`$BQ_PROJECT.$BQ_DATASET.observation\`
+         WHERE (observation_source_concept_id IN (903581, 1384450)
+            OR observation_source_concept_id IN (
+           SELECT DISTINCT concept_id
+           FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\` c
+           JOIN (
+             SELECT CAST(id AS STRING) AS id
+             FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\`
+             WHERE concept_id IN (1586134)
+               AND domain_id = 'SURVEY' ) a ON (c.path LIKE CONCAT('%', a.id, '.%'))
+           WHERE domain_id = 'SURVEY'
+             AND type = 'PPI'
+             AND subtype = 'QUESTION'
+         ))
+         AND questionnaire_response_id IS NULL
+
+           EXCEPT DISTINCT
+
+         SELECT DISTINCT person_id
+         FROM \`$BQ_PROJECT.$BQ_DATASET.observation\`
+         WHERE (observation_source_concept_id IN (903581, 1384450)
+            OR observation_source_concept_id IN (
+           SELECT DISTINCT concept_id
+           FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\` c
+           JOIN (
+             SELECT CAST(id AS STRING) AS id
+             FROM \`$BQ_PROJECT.$BQ_DATASET.cb_criteria\`
+             WHERE concept_id IN (1586134)
+               AND domain_id = 'SURVEY' ) a ON (c.path LIKE CONCAT('%', a.id, '.%'))
+           WHERE domain_id = 'SURVEY'
+             AND type = 'PPI'
+             AND subtype = 'QUESTION'
+         ))
+         AND questionnaire_response_id IS NOT NULL
+     )
+     AND (observation_source_concept_id IN (903581, 1384450)
```

Comment:
these concept ids represent some deprecated questions on the basics survey

Reply:
LGTM